src/Microsoft.Performance.SDK.Runtime/Discovery/AssemblyExtensionDiscovery.cs

```diff
@@ -203,70 +238,177 @@ public bool ProcessAssemblies(IEnumerable<string> directoryPaths)
                             var assembly = this.assemblyLoader.LoadAssembly(filePath);
                             if (!(assembly is null))
                             {
-                                allLoaded = allLoaded && ProcessAssembly(assembly);
-                            } else
+                                if (!ProcessAssembly(assembly, out var assemblyError))
+                                {
+                                    allLoaded = false;
+                                    Debug.Assert(assemblyError != null);
+                                    assemblyErrors.Add(assemblyError);
+                                }
+                            }
+                            else
                             {
+                                assemblyErrors.Add(
+                                    new ErrorInfo(ErrorCodes.AssemblyLoadFailed, "The assembly failed to load.")
```

Comment:
>ErrorCodes.AssemblyLoadFailed, "The assembly failed to load." [](start = 50, length = 61)

The descriptions seem very obvious based on the error codes in most (all?) cases. If we want both, should we have a static dictionary rather than adding the description to the ErrorInfo?